dali/pipeline/operators/reader/loader/loader.h

```diff
@@ -95,6 +107,11 @@ class Loader {
 
   // Get a random read sample
   LoadTargetPtr ReadOne() {
+    std::call_once(metadata_preparation_flag_, [this](){
+      if (lazy_init_) {
```

Comment:
I wonder if it is better to have this check outside.
If we do eager init, it is probably cheaper to have `if (lazy_init_)` first, as std::once_flag with std::call_once is probably using some synchronization.

Reply:
Right

Reply:
Done

Reply:
Done